00-programming-fundamentals/setting-up-vs-code-debugger.md

```diff
@@ -0,0 +1,112 @@
+  puts "a = #{a}"
+  puts "b = #{b}"
+
+  #completely necessary if-else
+  if a % 2 == 0
+    return add_them(a, b)
+  else
+    b += 2
+    return subtract_them(a, b)
+  end
+end
+
+#adds two numbers
+def add_them(a, b)
+  return a + b
+end
+
+#subtracts b from a
+def subtract_them(a, b)
+  return a - b
+end
+
+
+left_num = 5
+right_num = 3
+
+sum = cool_sum_func(left_num, right_num)
+
+puts sum
+```
+
+This is some very dopey code, but it will show us a lot about how the debugger gives us information.
+
+## Get Relevant Gems and Extensions
+If you have not already run the following:
+
+```bash
+$ gem install ruby-debug-ide
+$ gem install debase
+```
+
+Please run them now.
+
+## Configuring the Debugger
+
+VS Code, being language agnostic, doesn't have an internally implemented script that knows how to launch, run and hook into every single kind of program. Instead, it talks to other scripts and tools like the gems we just downloaded, and has generic functionality which those external tools can hook into. When we set up projects in VS Code, we create a series of configurations that VS Code stores in a file called `launch.json` in a folder called `.vscode`. However, if we set our project up this way, we would need to add that `.vscode` folder to every project we ever make. Nobody would do it. Instead, we will set up a global set of configurations in our VS Code settings.
+
+Open the debugger by using `⇧⌘D`, and look near the top left of the viewport to find a green triangle button that likely reads "No Configurations". When we are done, this list will be filled in with the names of our launch options.
+
+Using `⌘,` or  the drop down menu as seen below, open the settings menu.
+
+![Drop down Code, drop down Preferences, click Settings.](images/debug_settings_1.png "Drop down Code, drop down Preferences, click Settings.")
+
+When you have opened the settings, the bar at the top will have a curly brace icon as seen below. Clicking on it will open your `settings.json` for VS Code.
+
+![The settings page has a pair of curly braces to the right {}.](images/debug_settings_2.png "The settings page has a pair of curly braces to the right {}.")
+
+The code below represents a way in which we can launch our debugger on a ruby file that we have in focus, i.e. the last ruby file that you clicked on. Copy the text and paste it into your `settings.json` so that the launch keyword is at the same 'level' as the rest of your settings.
+
+```json
+"launch":{
+  "version": "0.2.0",
+  "configurations": [
+    {
+      "name": "Debug Open Ruby File",
+      "type": "Ruby",
+      "request": "launch",
+      "cwd": "${workspaceRoot}",
+      "program": "${file}"
+    }
+  ]
+}
+```
+
+when you are finished, it should look approximately like this:
+
+![The functional part of the json is in a field called 'launch' within the settings json.](images/debug_settings_3.png "The functional part of the json is in a field called 'launch' within the settings json")
```

Comment:
I might highlight the comma on the line before `"launch": {`, just because someone unfamiliar with JSON might mistake it.